src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -76,11 +76,7 @@ public static void VerifyAndUpdateFrameworkTypes(this IEnumerable<Schema> allSch
                     // Do not use property.SerializedName=="type" so that we can still use x-ms-client-name to override the auto-renaming here if there is some edge case.
                     if (property.Language.Default.Name.Equals("type", StringComparison.OrdinalIgnoreCase))
                     {
-                        // TODO: we may reuse the IsResourceModel method from SchemaExtension, but the result for flattened properties is different as although models with matched flattened properties are not treated as Resource but they still inherit from ResourceData. Here we align with the inheritance match to include flattened properties for match.
-                        if (property.Schema.Type == AllSchemaTypes.String
-                            && objSchema.ContainsStringProperty("id")
-                            && objSchema.ContainsStringProperty("name")
-                            || objSchema.Language.Default.Name.Contains("NameAvailability", StringComparison.Ordinal))
+                        if (objSchema.IsResourceData() || objSchema.Language.Default.Name.Contains("NameAvailability", StringComparison.Ordinal))
```

Comment:
A minor comment is that: will it be safer if we call `CSharpName()` function when we are checking the name?
```

Reply:
Updated.